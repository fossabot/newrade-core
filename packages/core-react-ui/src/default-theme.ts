/**
 *
 * Export the default theme values
 *
 */

export * from './default-theme/default-animations';
export * from './default-theme/default-buttons';
export * from './default-theme/default-code-colors';
export * from './default-theme/default-code-colors-dark';
export * from './default-theme/default-colors';
export * from './default-theme/default-colors-dark';
export * from './default-theme/default-effects';
export * from './default-theme/default-fonts';
export * from './default-theme/default-iconography';
export * from './default-theme/default-layout';
export * from './default-theme/default-sizing';
export * from './default-theme/default-typography';
