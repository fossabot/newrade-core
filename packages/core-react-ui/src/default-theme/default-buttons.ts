import { BorderStyles, Buttons } from '@newrade/core-design-system';

import { defaultColorIntents, defaultColors } from './default-colors';

export const defaultButtonBorder: BorderStyles = {
  default: {
    width: 1,
    style: 'solid',
    color: defaultColors.colorIntents.primary,
    radius: {
      all: 3,
    },
  },
  top: {},
  right: {},
  bottom: {},
  left: {},
};

export const defaultButtons: Buttons = {
  sizes: {
    large: {
      padding: {
        default: NaN,
        left: 16,
        right: 16,
        top: NaN,
        bottom: NaN,
      },
      iconPadding: 12,
      iconOnlyPadding: 6,
      border: {
        ...defaultButtonBorder,
        default: {
          ...defaultButtonBorder.default,
          radius: {
            all: 4,
          },
        },
      },
      outline: {
        color: defaultColorIntents.primary,
        width: defaultButtonBorderStyle.width,
        radius: defaultButtonBorderStyle.radius,
        style: defaultButtonBorderStyle.style,
      },
      iconSize: 24,
      iconMargin: {
        left: 10,
        right: 10,
        top: NaN,
        bottom: NaN,
      },
      border: defaultButtonBorder,
      outline: {},
      iconSize: 16,
      iconMargin: {},
    },
    medium: {
      padding: {
        default: NaN,
        left: 10,
        right: 10,
        top: NaN,
        bottom: NaN,
      },
      iconPadding: 10,
      iconOnlyPadding: 6,
      border: {
        ...defaultButtonBorder,
        default: {
          ...defaultButtonBorder.default,
          radius: {
            all: 4,
          },
        },
      },
      outline: {
        color: defaultColorIntents.primary,
        width: defaultButtonBorderStyle.width,
        radius: defaultButtonBorderStyle.radius,
        style: defaultButtonBorderStyle.style,
      },
      border: defaultButtonBorder,
      outline: {},
      iconSize: 16,
      iconMargin: {},
    },
    small: {
      padding: {
        default: NaN,
        left: 8,
        right: 8,
        top: NaN,
        bottom: NaN,
      },
      iconPadding: 8,
      iconOnlyPadding: 6,
      border: {
        ...defaultButtonBorder,
        default: {
          ...defaultButtonBorder.default,
          radius: {
            all: 3,
          },
        },
      },
      outline: {
        color: defaultColorIntents.primary,
        width: defaultButtonBorderStyle.width,
        radius: defaultButtonBorderStyle.radius,
        style: defaultButtonBorderStyle.style,
      },
      border: defaultButtonBorder,
      outline: {},
      iconSize: 16,
      iconMargin: {},
    },
    xSmall: {
      padding: {
        default: NaN,
        left: 6,
        right: 6,
        top: NaN,
        bottom: NaN,
      },
      iconPadding: 6,
      iconOnlyPadding: 6,
      border: {
        ...defaultButtonBorder,
        default: {
          ...defaultButtonBorder.default,
          radius: {
            all: 3,
          },
        },
      },
      outline: {
        color: defaultColorIntents.primary,
        width: defaultButtonBorderStyle.width,
        radius: defaultButtonBorderStyle.radius,
        style: defaultButtonBorderStyle.style,
      },
      border: defaultButtonBorder,
      outline: {},
      iconSize: 16,
      iconMargin: {},
    },
  },
  variants: {
    primary: {
      textColor: defaultColorIntents.primaryText,
      backgroundColor: defaultColorIntents.primaryText,
      iconColor: defaultColorIntents.primaryText,
      borderColor: defaultColorIntents.primaryText,
      outlineColor: defaultColorIntents.primaryText,
    },
    secondary: {
      textColor: defaultColorIntents.primaryText,
      backgroundColor: defaultColorIntents.primaryText,
      iconColor: defaultColorIntents.primaryText,
      borderColor: defaultColorIntents.primaryText,
      outlineColor: defaultColorIntents.primaryText,
    },
    tertiary: {
      textColor: defaultColorIntents.primaryText,
      backgroundColor: defaultColorIntents.primaryText,
      iconColor: defaultColorIntents.primaryText,
      borderColor: defaultColorIntents.primaryText,
      outlineColor: defaultColorIntents.primaryText,
    },
    positive: {
      textColor: defaultColorIntents.primaryText,
      backgroundColor: defaultColorIntents.primaryText,
      iconColor: defaultColorIntents.primaryText,
      borderColor: defaultColorIntents.primaryText,
      outlineColor: defaultColorIntents.primaryText,
    },
    warning: {
      textColor: defaultColorIntents.warningText,
      backgroundColor: defaultColorIntents.primaryText,
      iconColor: defaultColorIntents.primaryText,
      borderColor: defaultColorIntents.primaryText,
      outlineColor: defaultColorIntents.primaryText,
    },
    danger: {
      textColor: defaultColorIntents.dangerText,
      backgroundColor: defaultColorIntents.primaryText,
      iconColor: defaultColorIntents.primaryText,
      borderColor: defaultColorIntents.primaryText,
      outlineColor: defaultColorIntents.primaryText,
    },
  },
};

/**
 * DefaultButtons object without the `vars` property which is used only
 * in application-defined theme configuration
 */
export type DefaultButtons = Omit<Buttons, 'vars'>;
