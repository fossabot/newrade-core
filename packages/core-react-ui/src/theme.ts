/**
 *
 * Export the default theme CSS version (variables and default theme values)
 *
 */

export * from './theme/default-css-buttons';
export * from './theme/default-css-colors';
export * from './theme/default-css-sizing';
export * from './theme/default-theme.css';
