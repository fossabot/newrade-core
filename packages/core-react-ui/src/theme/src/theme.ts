/**
 *
 * Export the default theme vars (for Vanilla-extract styling files)
 *
 */

export * from './default-theme.css.js';
