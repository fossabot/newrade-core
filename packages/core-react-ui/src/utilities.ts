export * from './utilities/colors.utilities';
export * from './utilities/component.utilities';
export * from './utilities/css-variable.utilities';
export * from './utilities/effects.utilities';
export * from './utilities/filter-color.utility';
export * from './utilities/log.utilities';
export * from './utilities/text-capsize.utilities';
export * from './utilities/text.utilities';
export * from './utilities/theme.utilities';
export * from './utilities/utilities';

