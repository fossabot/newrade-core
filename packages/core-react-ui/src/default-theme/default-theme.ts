import { COLOR_SCHEME, DesignSystemV2 } from '@newrade/core-design-system';

import { defaultAnimations } from './default-animations';
import { defaultButtons } from './default-buttons';
import { defaultColors } from './default-colors';
import { defaultEffects } from './default-effects';
import { defaultIconography } from './default-iconography';
import { defaultLayout } from './default-layout';
import { defaultSizing } from './default-sizing';
import { defaultTypography } from './default-typography';

export const defaultTheme: DesignSystemV2 = {
  name: 'core-react-ui',
  colorScheme: COLOR_SCHEME.LIGHT,
  colors: defaultColors,
  effects: defaultEffects,
  sizing: defaultSizing,
  iconography: defaultIconography,
  typography: defaultTypography,
  layout: defaultLayout,
  animations: defaultAnimations,
  components: {
    buttons: defaultButtons,
  },
};
