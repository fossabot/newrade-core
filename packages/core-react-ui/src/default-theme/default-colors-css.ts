import { getCSSColorsV2 } from '../utilities-theme/colors.utilities';

import { defaultColors } from './default-colors';

export const defaultCSSColors = getCSSColorsV2(defaultColors);
