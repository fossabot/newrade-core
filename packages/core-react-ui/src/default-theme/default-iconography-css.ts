import { defaultIconography } from '../default-theme';
import { getCSSIconography } from '../utilities-theme/iconography.utilities';

export const defaultCSSIconography = getCSSIconography(defaultIconography);
