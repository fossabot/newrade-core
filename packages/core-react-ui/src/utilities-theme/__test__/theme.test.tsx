import { defaultTheme } from '../../default-theme/default-theme';
import { createCSSTheme } from '../theme.utilities';

describe('theme utilities', () => {
  describe(`${createCSSTheme.name}`, () => {
    it('should create a valid CSS theme', () => {
      const defaultCSSTheme = createCSSTheme(defaultTheme);
      expect(defaultCSSTheme.name).toMatchSnapshot();
      expect(defaultCSSTheme.colorScheme).toMatchSnapshot();
    });
  });
});
