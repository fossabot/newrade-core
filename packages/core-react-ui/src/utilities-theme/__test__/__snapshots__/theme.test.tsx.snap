// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`theme utilities createCSSTheme should create a valid CSS theme 1`] = `"core-react-ui"`;

exports[`theme utilities createCSSTheme should create a valid CSS theme 2`] = `"LIGHT"`;
